db/backup.go

```diff
@@ -69,6 +69,19 @@ func (db *db) basicImport(ctx context.Context, txn datastore.Txn, filepath strin
 				return NewErrJSONDecode(err)
 			}
 
+			// check if self referencing and remove from docMap for key creation
+			resetMap := map[string]any{}
```

Comment:
todo: This does not look like it will work with `TestBackupSelfRefImport_SplitPrimaryRelationWithSecondCollection_NoError` - your `TestBackupImport_DoubleReletionshipWithUpdate_NoError` test has defined the primary relations all on one object meaning it is not really self-referencing, although it is a gap I have missed in my PR and should be kept.

Either way I think you should wait for that PR to be merged in order to make use of the tests that flagged this error to be sure it  is fixed.

Reply:
It won't but it's a different problem. The error in `TestBackupSelfRefImport_SplitPrimaryRelationWithSecondCollection_NoError` is on export as explained in the note I left on your PR.

Reply:
(noted over there, but repeating here for ease of reference)

Was a really good spot by you there, but I still don't think the import code will work once the export is fixed.